src/main/java/com/google/cloud/spanner/pgadapter/statements/DdlExecutor.java

```diff
@@ -171,180 +52,33 @@ static String unquoteIdentifier(String identifier) {
   }
 
   /**
-   * Executes the given DDL statement. This can include translating the statement into a check
-   * whether the to create/drop object exists, and skipping the actual execution of the DDL
-   * statement depending on the result of the `if [not] exists` check.
+   * Executes the given DDL statement. This can also include some translations that are needed to
+   * make the statement compatible with Cloud Spanner.
    */
   StatementResult execute(ParsedStatement parsedStatement, Statement statement) {
-    Statement translated = translate(parsedStatement, statement);
-    if (translated != null) {
-      return connection.execute(translated);
-    }
-    return NOT_EXECUTED;
+    return connection.execute(translate(parsedStatement, statement));
   }
 
   /**
-   * Translates a DDL statement that can contain an `if [not] exists` statement into one that is
-   * supported by the backend.
+   * Translates a DDL statement that can contain 'create table' statement with a named primary key
+   * into one that is supported by the backend.
    */
   Statement translate(ParsedStatement parsedStatement, Statement statement) {
     SimpleParser parser = new SimpleParser(parsedStatement.getSqlWithoutComments());
     if (parser.eatKeyword("create")) {
       statement = translateCreate(parser, statement);
-    } else if (parser.eatKeyword("drop")) {
-      statement = translateDrop(parser, statement);
     }
 
     return statement;
   }
 
   Statement translateCreate(SimpleParser parser, Statement statement) {
     if (parser.eatKeyword("table")) {
-      Statement createTableStatement = translateCreateTable(parser, statement);
-      if (createTableStatement == null) {
-        return null;
-      }
-      return maybeRemovePrimaryKeyConstraintName(createTableStatement);
-    }
-    boolean unique = parser.eatKeyword("unique");
-    if (parser.eatKeyword("index")) {
-      return translateCreateIndex(parser, statement, unique);
-    }
-    return statement;
-  }
-
-  Statement translateDrop(SimpleParser parser, Statement statement) {
```

Comment:
Just to confirm, we are removing translateDrop because IF exist translation is no longer required?

Reply:
Correct, and we also don't have any other translations for `DROP` statements.

(For comparison: For `CREATE` statements we don't drop the entire translate method, because we still translate `CREATE TABLE` statements that include a named primary key, when that primary key just uses the same name as what Cloud Spanner would otherwise generate)